### PR TITLE
[ENH] Enforced string column names by default when using clean_names.

### LINKED ISSUE #612: [ENH] Proposal: Enforce automagic column name conversion to string type in `clean_names`
UPDATE FROM MAINTAINERS: ANYBODY WHO IS INTERESTED IN THIS ISSUE, PLEASE SEE [THIS COMMENT](https://github.com/ericmjl/pyjanitor/issues/612#issuecomment-557250963) FOR PROPOSED CODE.

------

# Brief Description

`clean_names` method does not work when an integer is used as column name 

# Minimally Reproducible Code

```
rankings = {
    "countries_to_play_cricket": [
        "India",
        "South Africa",
        "England",
        "New Zealand",
        "Australia",
    ],
    0: ["England", "India", "New Zealand", "South Africa", "Pakistan"],
    "t20": ["Pakistan", "India", "Australia", "England", "New Zealand"],
}

rankings_pd = pd.DataFrame(rankings)

df = rankings_pd.clean_names()
```

# Error Messages

```
AttributeError: 'int' object has no attribute 'lower'
```

If I change the column name integer 0 to string "0" then there will be no error. But in that case it defeats the purpose of using a method :)

### PR DESCRIPTION
**Pull request by ericmjl**
# PR Description

Please describe the changes proposed in the pull request:

- Enforces string type for column names when calling `clean_names()`.
- Coercion is on by default; can be turned off by a flag.
- Docstring documents the behaviour of the function.

**This PR resolves #612.** 

# Relevant Reviewers

Happy to take comments from anybody tagged here :smile::

- @rahosbach 
- @zbarry 
- @szuckerman 
- @jk3587 
- @sallyhong 
- @shandou 
- @hectormz

### CODE REVIEW

File: janitor/functions.py [hunk 1]
```diff
@@ -17,6 +17,8 @@
     Tuple,
     Union,
 )
+from types import SimpleNamespace
```
Comment (hectormz) [inline janitor/functions.py]: Is this import related to the current PR?

Comment (ericmjl) [reply janitor/functions.py]: Good catch!

Comment (ericmjl) [reply janitor/functions.py]: This was an unused import.
